common/shared.c

```diff
@@ -213,123 +213,25 @@ static void ft_cntr_set_wait_attr(void)
 	}
 }
 
-int ft_cntr_open(struct fid_cntr **cntr)
-{
-	ft_cntr_set_wait_attr();
-	return fi_cntr_open(domain, &cntr_attr, cntr, cntr);
-}
-
-int ft_rma_read_target_allowed(uint64_t caps)
-{
-	if (caps & (FI_RMA | FI_ATOMIC)) {
-		if (caps & FI_REMOTE_READ)
-			return 1;
-		return !(caps & (FI_READ | FI_WRITE | FI_REMOTE_WRITE));
-	}
-	return 0;
-}
-
-int ft_rma_write_target_allowed(uint64_t caps)
-{
-	if (caps & (FI_RMA | FI_ATOMIC)) {
-		if (caps & FI_REMOTE_WRITE)
-			return 1;
-		return !(caps & (FI_READ | FI_WRITE | FI_REMOTE_WRITE));
-	}
-	return 0;
-}
-
 uint64_t ft_info_to_mr_access(struct fi_info *info)
 {
 	uint64_t mr_access = 0;
-	if ((info->mode & FI_LOCAL_MR) || (info->domain_attr->mr_mode & FI_MR_LOCAL)) {
-		if (info->caps & (FI_MSG | FI_TAGGED)) {
-			if (info->caps & FT_MSG_MR_ACCESS) {
-				mr_access |= info->caps & FT_MSG_MR_ACCESS;
-			} else {
-				mr_access |= FT_MSG_MR_ACCESS;
-			}
-		}
 
-		if (info->caps & (FI_RMA | FI_ATOMIC)) {
-			if (info->caps & FT_RMA_MR_ACCESS) {
-				mr_access |= info->caps & FT_RMA_MR_ACCESS;
-			} else	{
-				mr_access |= FT_RMA_MR_ACCESS;
-			}
-		}
-	} else {
-		if (info->caps & (FI_RMA | FI_ATOMIC)) {
-			if (ft_rma_read_target_allowed(info->caps)) {
-				mr_access |= FI_REMOTE_READ;
-			}
-			if (ft_rma_write_target_allowed(info->caps)) {
-				mr_access |= FI_REMOTE_WRITE;
-			}
-		}
-	}
-	return mr_access;
-}
-
-#define bit_isset(x, i) (x >> i & 1)
-#define for_each_bit(x, i) for (i = 0; i < (8 * sizeof(x)); i++)
-
-static inline int bit_set_count(uint64_t val)
-{
-	int cnt = 0;
-	while (val) {
-		cnt++;
-		val &= val - 1;
-	}
-	return cnt;
-}
-
-int ft_alloc_bit_combo(uint64_t fixed, uint64_t opt,
-		       uint64_t **combos, int *len)
-{
-	uint64_t *flags;
-	int i, num_flags;
-	uint64_t index;
-	int ret;
-
-	num_flags = bit_set_count(opt) + 1;
-	flags = calloc(num_flags, sizeof(fixed));
-	if (!flags) {
-		perror("calloc");
-		return -FI_ENOMEM;
-	}
-
-	*len = 1 << (num_flags - 1);
-	*combos = calloc(*len, sizeof(fixed));
-	if (!(*combos)) {
-		perror("calloc");
-		ret = -FI_ENOMEM;
-		goto clean;
-	}
-
-	num_flags = 0;
-	for_each_bit(opt, i) {
-		if (bit_isset(opt, i))
-			flags[num_flags++] = 1ULL << i;
+	if ((info->mode & FI_LOCAL_MR) && (info->caps & (FI_MSG | FI_TAGGED))) {
+		if (info->caps & FT_MSG_MR_ACCESS)
+			mr_access |= info->caps & FT_MSG_MR_ACCESS;
+		else
+			mr_access |= FT_MSG_MR_ACCESS;
 	}
 
-	for (index = 0; index < (*len); index++) {
-		(*combos)[index] = fixed;
-		for_each_bit(index, i) {
-			if (bit_isset(index, i))
-				(*combos)[index] |= flags[i];
-		}
+	if (info->caps & (FI_RMA | FI_ATOMIC)) {
+		if (info->caps & FT_RMA_MR_ACCESS)
+			mr_access |= info->caps & FT_RMA_MR_ACCESS;
+		else
+			mr_access |= FT_RMA_MR_ACCESS;
 	}
-	ret = FI_SUCCESS;
 
-clean:
-	free(flags);
-	return ret;
-}
-
-void ft_free_bit_combo(uint64_t *combo)
-{
-	free(combo);
+	return mr_access;
 }
 
```

Comment:
@a-ilango in 4th commit, all missing stuff was added.
I'm going to split each added functionality into separate commit tomorrow.